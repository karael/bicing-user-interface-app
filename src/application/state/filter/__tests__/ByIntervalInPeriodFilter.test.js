import ByIntervalInPeriodFilter from 'application/state/filter/ByIntervalInPeriodFilter';

describe('application/state/filter/ByIntervalInPeriodFilter', () => {
  test('it can create a ByIntervalInPeriodFilter from raw values', () => {
    const periodStart = '2017-08-12 13:00:30'; const periodEnd = '2017-09-02 13:00:30'; const
      interval = '5 min';

    const filter = ByIntervalInPeriodFilter.fromRawValues(periodStart, periodEnd, interval);

    expect(filter.periodStart).toEqual(periodStart);
    expect(filter.periodEnd).toEqual(periodEnd);
    expect(filter.interval).toEqual(interval);
  });
  test('it can return a zoom value from a ByIntervalInPeriodFilter if periodStart is not a date', () => {
    expect(() => ByIntervalInPeriodFilter.fromRawValues('not date format', '2017-09-02 13:00:30', '5 min')).toThrowErrorMatchingSnapshot();
  });
  test('it can return a zoom value from a ByIntervalInPeriodFilter if periodEnd is not a date', () => {
    expect(() => ByIntervalInPeriodFilter.fromRawValues('2017-09-02 13:00:30', 'not date format', '5 min')).toThrowErrorMatchingSnapshot();
  });
  test('it can return a zoom value from a ByIntervalInPeriodFilter if interval is not a string', () => {
    expect(() => ByIntervalInPeriodFilter.fromRawValues('2017-09-02 13:00:30', '2017-10-02 13:00:30', 12345)).toThrowErrorMatchingSnapshot();
  });
  test('it can return a zoom value from a ByIntervalInPeriodFilter if periodStart greater than periodEnd ', () => {
    expect(() => ByIntervalInPeriodFilter.fromRawValues('2019-09-02 13:00:30', '2017-10-02 13:00:30', '5 min')).toThrowErrorMatchingSnapshot();
  });
});
