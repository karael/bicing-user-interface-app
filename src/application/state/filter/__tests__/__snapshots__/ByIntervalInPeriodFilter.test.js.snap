// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`application/state/filter/ByIntervalInPeriodFilter it can return a zoom value from a ByIntervalInPeriodFilter if interval is not a string 1`] = `
"{
  \\"periodStart\\": \\"2017-09-02 13:00:30\\",
  \\"periodEnd\\": \\"2017-10-02 13:00:30\\",
  \\"interval\\" [31m[1][0m: 12345
}
[31m
[1] \\"interval\\" must be a string[0m"
`;

exports[`application/state/filter/ByIntervalInPeriodFilter it can return a zoom value from a ByIntervalInPeriodFilter if periodEnd is not a date 1`] = `
"{
  \\"periodStart\\": \\"2017-09-02 13:00:30\\",
  \\"interval\\": \\"5 min\\",
  \\"periodEnd\\" [31m[1][0m: \\"not date format\\"
}
[31m
[1] \\"periodEnd\\" must be a number of milliseconds or valid date string[0m"
`;

exports[`application/state/filter/ByIntervalInPeriodFilter it can return a zoom value from a ByIntervalInPeriodFilter if periodStart greater than periodEnd  1`] = `
"{
  \\"periodStart\\": \\"2019-09-02 13:00:30\\",
  \\"interval\\": \\"5 min\\",
  \\"periodEnd\\" [31m[1][0m: \\"2017-10-02 13:00:30\\"
}
[31m
[1] \\"periodEnd\\" must be greater than \\"Mon Sep 02 2019 13:00:30 GMT+0000 (UTC)\\"[0m"
`;

exports[`application/state/filter/ByIntervalInPeriodFilter it can return a zoom value from a ByIntervalInPeriodFilter if periodStart is not a date 1`] = `"Invalid date format"`;
